### PR TITLE
add new page to handle awaiting-pronouncement with date of hearing

### PR DESCRIPTION
**Pull request by dtaylor7**
# Description

[DIV-4118 - Respondent can see the details of the DN approval in their account](https://tools.hmcts.net/jira/browse/DIV-4118)
 - add new page to handle awaiting-pronouncement with date of hearing
 - Integrate document handler module

### CODE REVIEW

File: steps/respondent/progress-bar/ProgressBar.content.json [hunk 1]
```diff
@@ -5,6 +5,11 @@
     "youMustRespond": "You must respond",
     "getDecreeNisi": "Your {{ session.divorceWho }} gets a decree nisi",
     "madeFinal": "Divorce is complete",
+    "downloadableFiles": "Download your documents",
+    "files": {
+      "dpetition": "Divorce application",
+      "aosinvitation": "Entitlement to a decree"
```
Comment (ravinderpal) [inline steps/respondent/progress-bar/ProgressBar.content.json]: `"entitlementToDecree": "Entitlement to a decree"` maybe?

 "aosinvitation" is a completely different things afaik

File: config/custom-environment-variables.yml [hunk 2]
```diff
@@ -62,6 +62,8 @@ services:
     baseUrl: FEES_AND_PAYMENTS_URL
     health: FEES_AND_PAYMENTS_HEALTHCHECK_URL
   proxyUrl: PROXY_URL
+  evidenceManagment:
```
Comment (ravinderpal) [inline config/custom-environment-variables.yml]: *evidenceManagement

File: config/default.yml [hunk 3]
```diff
@@ -62,6 +62,8 @@ services:
     baseUrl: http://localhost:3001
     health: http://localhost:3001/health
   proxyUrl: ''
+  evidenceManagment:
```
Comment (ravinderpal) [inline config/default.yml]: again *evidenceManagement
I think there an 'e' missing in every 'management' word

File: infrastructure/main.tf [hunk 4]
```diff
@@ -12,6 +12,7 @@ locals {
   div_cos_url              = "http://div-cos-${local.local_env}.service.core-compute-${local.local_env}.internal"
   div_cms_url              = "http://div-cms-${local.local_env}.service.core-compute-${local.local_env}.internal"
   div_fps_url              = "http://div-fps-${local.local_env}.service.core-compute-${local.local_env}.internal"
+  div_emca_url             = "http://div-emca-${local.local_env}.service.core-compute-${local.local_env}.internal"
```
Comment (ravinderpal) [inline infrastructure/main.tf]: where is this used and what it is used for?

File: steps/respondent/progress-bar/ProgressBar.content.json (hunk 1 above)
Comment (dtaylor7) [reply steps/respondent/progress-bar/ProgressBar.content.json]: this is the file name....

File: config/custom-environment-variables.yml [hunk 5]
```diff
@@ -62,6 +62,8 @@ services:
     baseUrl: FEES_AND_PAYMENTS_URL
     health: FEES_AND_PAYMENTS_HEALTHCHECK_URL
   proxyUrl: PROXY_URL
+  evidenceManagement:
+    downloadFileUrl: EVIDENCE_MANAGMENT_SERVICE_FILE_DOWNLOAD_URL
```
Comment (hosainnet) [inline config/custom-environment-variables.yml]: We probably want to split this into a BASE and a HEALTHCHECK env variables, and also add it to https://github.com/hmcts/div-respondent-frontend/blob/master/middleware/healthcheck.js#L22

Comment (dtaylor7) [reply config/custom-environment-variables.yml]: why?

Comment (hosainnet) [reply config/custom-environment-variables.yml]: Each service exposes `/health` to see if the service can successfully contact its dependencies
`https://div-rfe-aat.service.core-compute-aat.internal/health`

Comment (dtaylor7) [reply config/custom-environment-variables.yml]: ahh yes... need to healthcheck...

File: kubernetes/config.aat.yaml [hunk 6]
```diff
@@ -31,3 +31,5 @@ data:
   CASE_MAINTENANCE_BASE_URL: "http://div-cms-aat.service.core-compute-aat.internal"
   LOG_LEVEL: "DEBUG"
   PROXY_URL: "http://proxyout.reform.hmcts.net:8080"
+  EVIDENCE_MANAGMENT_SERVICE_URL: "http://div-emca-aat.service.core-compute-aat.internal"
```
Comment (ravinderpal) [inline kubernetes/config.aat.yaml]: *MANAGEMENT typo here

File: kubernetes/config.aat.yaml [hunk 7]
```diff
@@ -31,3 +31,5 @@ data:
...
+  EVIDENCE_MANAGMENT_SERVICE_HEALTHCHECK_URL: "http://div-emca-aat.service.core-compute-aat.internal/health"
```
Comment (ravinderpal) [inline kubernetes/config.aat.yaml]: *MANAGEMENT typo here

File: infrastructure/main.tf [hunk 8]
```diff
@@ -118,6 +119,8 @@ module "frontend" {
     DECREE_NISI_FRONTEND_URL = "${var.decree_nisi_frontend_url}"
     FEES_AND_PAYMENTS_URL = "${local.div_fps_url}"
     FEES_AND_PAYMENTS_HEALTHCHECK_URL = "${local.div_fps_url}${var.health_endpoint}"
+    EVIDENCE_MANAGMENT_SERVICE_URL = "${local.div_emca_url}"
+    EVIDENCE_MANAGMENT_SERVICE_HEALTHCHECK_URL = "${local.div_emca_url}${var.health_endpoint}"
```
Comment (ravinderpal) [inline infrastructure/main.tf]: *MANAGEMENT typo here on L 122 & 123

File: mocks/services/case-orchestration/retrieve-aos-case/mock-decree-nisi-announcement.json [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+        "createdOn": null,
+        "lastModifiedBy": 0,
+        "modifiedOn": null,
+        "fileName": "aosinvitation1539017559370699",
+        "fileUrl": "http://dm-store-aat.service.core-compute-aat.internal/documents/a060ba13-d5da-45b0-a181-1c2ace891904",
+        "mimeType": null,
+        "status": null
+      }
+    ],
+    "D8DocumentsGenerated": [
```
Comment (ravinderpal) [inline mocks/services/case-orchestration/retrieve-aos-case/mock-decree-nisi-announcement.json]: should this be lower-camelCase like all the other case data fields?

File: config/custom-environment-variables.yml [hunk 10]
```diff
@@ -62,6 +62,9 @@ services:
     baseUrl: FEES_AND_PAYMENTS_URL
     health: FEES_AND_PAYMENTS_HEALTHCHECK_URL
   proxyUrl: PROXY_URL
+  evidenceManagement:
+    baseUrl: EVIDENCE_MANAGMENT_SERVICE_URL
```
Comment (jvictorcf) [inline config/custom-environment-variables.yml]: typo

**Pull request closed without merging**